src/components/components/AddLayerPanel/AddLayerPanel.component.jsx

```diff
@@ -0,0 +1,362 @@
+      icon: '',
+      description:
+        'Adds an entity to load and display 3d tiles from Google Maps Tiles API 3D Tiles endpoint. This will break your scene and you cannot save it yet, so beware before testing.',
+      id: 4,
+      handlerFunction: create3DTiles
+    },
+    {
+      name: 'Create custom model',
+      img: '',
+      icon: '',
+      description:
+        'Create entity with model from path for a glTF (or Glb) file hosted on any publicly accessible HTTP server.',
+      id: 5,
+      handlerFunction: createCustomModel
+    },
+    {
+      name: 'Create primitive geometry',
+      img: '',
+      icon: '',
+      description:
+        'Create entity with A-Frame primitive geometry. Geometry type could be changed in properties panel.',
+      id: 6,
+      handlerFunction: createPrimitiveGeometry
+    }
+  ];
+
+  // get array with objects data (cardsData) from mixinGroups of selectedOption
+  const getSelectedMixinCards = (selectedOption) => {
+    if (!selectedOption) return [];
+    const selectedOptionData = options.find(
+      (option) => option.value === selectedOption
+    );
+    const selectedMixinGroupNames = selectedOptionData.mixinGroups;
+
+    // there are no mixin groups
+    if (!selectedMixinGroupNames) return [];
+
+    // filter selected mixin groups from all mixin groups (groupedMixins)
+    const cardsData = groupedMixins
+      .filter((group) => selectedMixinGroupNames.includes(group.label))
+      .flatMap((mixinGroup) => mixinGroup.options);
+
+    return cardsData;
+  };
+
+  let selectedCards;
+  if (selectedOption == 'Layers: Streets & Intersections') {
+    selectedCards = layersData;
+  } else {
+    selectedCards = getSelectedMixinCards(selectedOption);
+  }
+
+  const handleSelect = (value) => {
+    setSelectedOption(value);
+  };
+
+  /* create and preview entity events */
+
+  // entity preview element
+  let preEntity = document.createElement('a-entity');
+  let cameraFrontVec = new THREE.Vector3();
+  let selectedObjPos = new THREE.Vector3();
+  let segmentElevationPosY = 0;
+  let segmentOfSelectedEl;
+
+  preEntity.setAttribute('visible', false);
+
+  AFRAME.scenes[0].appendChild(preEntity);
+
+  const previewEntity = () => {
+    if (selectedOption == 'Models: Buildings') {
+      preEntity.setAttribute('scale', { x: 0.1, y: 0.1, z: 0.1 });
+    } else {
+      preEntity.setAttribute('scale', { x: 0.5, y: 0.5, z: 0.5 });
+    }
+    /* place object in front of active camera */
+    const direction = new THREE.Vector3();
+    // active camera object
+    const activeCamera = AFRAME.INSPECTOR.camera;
+    activeCamera.getWorldDirection(direction);
+    cameraFrontVec.copy(activeCamera.position);
+    cameraFrontVec.add(direction.multiplyScalar(5));
+
+    // place preview entity in front of camera in 5 meters
+    preEntity.setAttribute('position', cameraFrontVec);
+  };
+
+  const getParentSegmentEl = (element) => {
+    if (element.className.includes('segment-parent')) {
```

Comment:
It depends of what you want to do here, I don't see segment-parent used anywhere so I'm not sure what is the intended behavior of this code.
If you want to test if the element has the class "segment-parent", it's preferable to do
`element.classList.contains('segment-parent'))`
The line `element.className.includes('segment-parent')` will match any class with segment-parent in it, like `prefix-segment-parent-suffix`.

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

Reply:
> The line element.className.includes('segment-parent') will match any class with segment-parent in it, like prefix-segment-parent-suffix.

Segment classes look like this: `segment-parent-1`, `segment-parent-2`, ... So I use `includes` method here.
https://github.com/3DStreet/3dstreet/blob/7017c3db1e3aff5bfe2e0167f12ed1325119984e/src/aframe-streetmix-parsers.js#L797

> It depends of what you want to do here, I don't see segment-parent used anywhere so I'm not sure what is the intended behavior of this code.

This is the code I use to find the parent segment for the selected element. To later get the `data-elevation-posY` attribute from `segment-parent` element, which contains the segment height value. This value is used to position the element to be added according to the height of the segment (elevation level)

Reply:
Thanks for the explanation, it makes sense. All good then!